baseapp/abci.go

```diff
@@ -920,3 +921,19 @@ func splitPath(requestPath string) (path []string) {
 
 	return path
 }
+
+// createQueryContext creates a new sdk.Context for a query, taking as args
+// the block height and whether the query needs a proof or not.
+func (app *BaseApp) createQueryContextWithCheckState() sdk.Context {
+
+	cacheMS := app.checkState.CacheMultiStore()
+
+	// branch the commit-multistore for safety
+	app.checkStateMtx.RLock()
+	ctx := sdk.NewContext(
+		cacheMS, app.checkState.ctx.BlockHeader(), true, app.logger,
+	).WithMinGasPrices(app.minGasPrices).WithBlockHeight(app.LastBlockHeight())
+	app.checkStateMtx.RUnlock()
```

Comment:
Can you move "app.checkStateMtx.RUnlock()" to right after "app.checkStateMtx.RLock()" with "defer" keyword.
If you move "defer app.checkStateMtx.RUnlock()" to L933, Mutex gets closed even if any exceptional case happens.

Reply:
Thank you for your comment but the logic between `RLock` is deterministic. So it's impossible to occur any exceptional cases which makes deadlock.

Reply:
Ok, I understand.